File: support_tools/python/upd_commander_checker/src/upd_commander_checker/report_output.py
from pathlib import Path


def finish_report(lines: list[str], output: str | None, exit_code: int) -> int:
    for line in lines:
        print(line)
    if not output:
        return exit_code

    path = Path(output)
    try:
        path.parent.mkdir(parents=True, exist_ok=True)
        path.write_text("\n".join(lines) + "\n", encoding="utf-8")
    except (OSError, ValueError):
        print(f"I/O ERROR: failed to write output: {path}")
        return 2
    return exit_code

File: support_tools/go/upd_commander_checker/cmd/upd-commander-check/report_output.go
package main

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"
)

func finishReport(lines []string, output string, code int) int {
	for _, line := range lines {
		fmt.Println(line)
	}
	if output == "" {
		return code
	}
	if dir := filepath.Dir(output); dir != "." {
		if err := os.MkdirAll(dir, 0o755); err != nil {
			fmt.Printf("I/O ERROR: failed to write output: %s\n", output)
			return 2
		}
	}
	if err := os.WriteFile(output, []byte(strings.Join(lines, "\n")+"\n"), 0o644); err != nil {
		fmt.Printf("I/O ERROR: failed to write output: %s\n", output)
		return 2
	}
	return code
}
